Source/WTF/wtf/text/ASCIILiteral.h

```diff
@@ -45,15 +45,23 @@ OBJC_CLASS NSString;
 namespace WTF {
 
 class PrintStream;
+class String;
 
 class ASCIILiteral final {
 public:
     constexpr operator const char*() const { return m_charactersWithNullTerminator.data(); }
 
+    static constexpr ASCIILiteral fromNullTerminatedStringUnsafe(const char* string)
```

Comment:
I don't think this makes sense as an addition to ASCIILiteral because it violates the lifetime model that ASCIILiteral points to static storage. 

In the ideal world, we'll remove operator _s and it will be a compile time error to construct ASCIILiteral without static storage.